File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0;
        int right = heights.length - 1;
        int max = 0;

        while(left < right) {
            int minHeight = Math.min(heights[left], heights[right]);
            max = Math.max(max, minHeight * (right-left));

            if (heights[left] <= heights[right]) left++;
            else right--;
        }

        return max;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */

    findMin(nums) {
        let left = 0,
            right = nums.length-1;

        while (left <= right) {
            const mid = Math.floor((left+right)/2);

            if (nums[left] <= nums[mid] && nums[left] <= nums[right]) return nums[left]; //fully sorted

            if (mid > 0 && nums[mid-1] > nums[mid]) return nums[mid];

            if (nums[mid] > nums[right]) left = mid+1;
            else right = mid-1;
        }
    }
}
